### DISCUSSION TITLE
Which parser should be supported next?

### DISCUSSION THREAD

**Discussion by MH4GF** [liam-hq/liam#364] [Announcements]
Currently, the following formats can be parsed and rendered ER diagrams

- PostgreSQL
- schema.rb(Ruby on Rails)
- schema.prisma(Prisma)
- tbls

ref: https://liambx.com/docs/parser/supported-formats

Which parser should be supported next? Let us know which ORM you would like to use!

**Comment by MH4GF** [comment]
Prisma

**Reply by MH4GF** [reply]
- In progress: https://github.com/liam-hq/liam/issues/506

**Reply by MH4GF** [reply]
Supported!
ref: https://liambx.com/docs/parser/supported-formats/prisma

**Comment by MH4GF** [comment]
Drizzle

**Reply by MH4GF** [reply]
- https://github.com/liam-hq/liam/issues/1293

**Comment by MH4GF** [comment]
MySQL

**Reply by MH4GF** [reply]
Suppoted via tbls.
ref: https://liambx.com/docs/parser/supported-formats/mysql

**Comment by MH4GF** [comment]
Goose

**Comment by Trubador** [comment]
- Mongo DB collections
- Microsoft SQL Server
- Redis?

**Reply by hoshinotsuyoshi** [reply]
Among these, `Microsoft SQL Server` is an RDBMS, making it the closest to what we already support and potentially the easiest to implement. I've created a ticket for it: https://github.com/liam-hq/liam/issues/681

**Reply by hoshinotsuyoshi** [reply]
> Among these, Microsoft SQL Server is an RDBMS, making it the closest to what we already support and potentially the easiest to implement. I've created a ticket for it: https://github.com/liam-hq/liam/issues/681

The ticket has been closed, and documentation has been added! 🎉 https://liambx.com/docs/parser/supported-formats/mssql

**Reply by salih-kv** [reply]
MongoDB (via Mongoose schemas) would be great, created a ticket for this: #4074

**Comment by YoussefRabeiii** [comment]
Drizzle ORM For sure its one of the best and have alot of potential

**Reply by MH4GF** [reply]
- https://github.com/liam-hq/liam/issues/1293

**Comment by banhmysuawx** [comment]
I think you should support more to **Django ORM**

**Reply by MH4GF** [reply]
@banhmysuawx If you are using PostsgreSQL, you can do that by doing a pg_dump. We have created a documentation page so you can check it out!

[Django ORM - Liam ERD](https://liambx.com/docs/parser/supported-formats/django)

**Reply by banhmysuawx** [reply]
@MH4GF Thanks for the in pointer to the docs and backing the project! Your `pg_dump` guidance is really useful I will read the docs in details. Thanks again!

**Comment by handfuloflight** [comment]
We need support for Convex schemas: https://docs.convex.dev/database/schemas

**Reply by YoussefRabeiii** [reply]
any news on this one

**Comment by ireneisdoomed** [comment]
We would highly appreciate support for the Croissant format, a machine learning metadata standard https://github.com/mlcommons/croissant

**Reply by ireneisdoomed** [reply]
I want to give this a push!! Croissant is used by HuggingFace, Google Datasets, and Kaggle.

**Comment by jmherbst** [comment]
https://sailsjs.com/ with mongodb

**Comment by manoj-lookhi** [comment]
Databricks Unity Catalog.

**Comment by SamRozhkov** [comment]
Greenplum and clickhouse

**Comment by ghost** [comment]
✅ **SQLite**

---

For example, rendering ER diagrams for Grist would be wonderful - https://github.com/gristlabs/grist-core

**Reply by MH4GF** [reply]
@fullstackdesign-xyz Thanks! SQLite can be visualized via tbls if it's the CLI version.
https://liambx.com/docs/parser/supported-formats/sqlite

But there was a problem with tbls and grist couldn't visualize it as is, I'm putting up an issue: https://github.com/k1LoW/tbls/issues/709

**Comment by KaranDrivesEvents** [comment]
Huge respect for what you’ve built. We’ve just launched an ER-diagram copilot and I’m interested in exploring whether our teams can find a collaborative angle that benefits both sides.

Karan Patel,
Founder/Problem Solver - TalkingSchema.ai

CC - @MH4GF

**Comment by gainings** [comment]
I request support for **DBML** as general usecase. This support to important to users of dbt and dbterd.
ref: https://dbml.dbdiagram.io/home/
ref: https://github.com/datnguye/dbterd?tab=readme-ov-file#-supported-output-formats

**Comment by YoussefRabeiii** [comment]
amazing work as always guys for real I LOVVVVE using liam with the latest Drizzle orm updates specially after having the ability to do entire folder at once i hope my pr helped, sooooo is there any chance or plan on supporting convex schema

**Reply by YoussefRabeiii** [reply]
@MH4GF is it planned to support convex 👀👀
